sherpa/plot/pylab_backend.py

```diff
@@ -586,7 +572,7 @@ def get_plot_defaults():
 
 
 def get_point_defaults():
-    return get_keyword_defaults(point, 2)
+    return {'marker': None, 'alpha': None, 'color': None}
```

Comment:
There are other places where we also hardcode definitions instead of pulling them all from `get_keyword_defaults`, so this is not unusual in the context of this backend.
(And, I think it's beneficial to get away from the `get_keyword_defaults` in general, because that risks polluting the general plot objects with backend-specific keywords, which makes backend switching hard, but that is a larger change not part of this PR.)

Reply:
For background info and not directly related to this PR:

Currently the `sherpa.ui.utils.Session` and `sherpa.astro.ui.utils.Session` class (which subclasses the former) - and are the logic between the `sherpa.ui`/`sherpa.astro.ui` modules - create plot objects

- when they are created
-  when their `clean` method is called

We could make it lazy - we only create the actual object when it is first needed - but there is still the issue to decide what the `Session` class should do when the plot backend is changed.

The session approach relies on access to these plot objects - and persisting those objects between calls - so unless the persistence is removed (which I am likely to argue against) it is hard to say we can always just create a new plot object each time (and hence pick up the new plot backend).

Now, I think we could argue that users of the session class (either directly or indirectly) are unlikely to swap plot backends, but I can imagine that I might want to swap between `matplotlib` and `bokeh` depending on whether I'm in a notebook or not. It also makes testing easier if we have a story for this case.

Reply:
The goal in #1191 is that make the plots objects work with any backend, so that we would not have to re-create the plot objects, just because the backend is changed; instead the backend is hot-swapable (if that's a word). Plot objects only call "backend.plot", but that "backend" is not a variable held in the object, it's a module level variable. So, you can change it and "backend" would suddenly refer to something else (say bokeh instead of matplotlib). One can think of it analogous to like unix pipes: The plot objects pipes into `backend.histo` and does not need to care what exactly that next step (`backend.histo`) does with the data that's send to it. In that senses, it's somewhat similar to monkey-patching, but not as dangerous (in monkey-patching I  would be changing methods of an existing object, while here I'm simply changing which object is getting the plotting command). So, I would hope that #1191 takes care of this concern. If it does not, that's an incompleteness of #1191 and I should add commits to that.

(With one important exception: Our default plot styling will only use options that all implemented backend understand such as "linestyle="solid", but it is possible to manually set preferences that are backend-specific. Those won't work when the backend is switched. But that's a design feature, not a bug. I will also supply, in the separate PR, a way to validate that all examples and all default plot settings in Sherpa limit itself to a well-defined set of options that all implemented backends understand.)